supernode/services/storagechallenge/actor.go

```diff
@@ -0,0 +1,164 @@
+
+import (
+	"github.com/AsynkronIT/protoactor-go/actor"
+	"github.com/pastelnetwork/gonode/common/context"
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/pastelnetwork/gonode/messaging"
+	dto "github.com/pastelnetwork/gonode/proto/supernode"
+	"github.com/pastelnetwork/gonode/supernode/node"
+	"google.golang.org/grpc/metadata"
+)
+
+type sendVerifyStorageChallengeMsg struct {
+	*messaging.CommonProtoMsg
+	Context                 context.Context
+	VerifierMasternodesAddr []string
+	*ChallengeMessage
+}
+
+func (s *sendVerifyStorageChallengeMsg) getAppContext() context.Context {
+	if s.Context == nil {
+		return context.Background()
+	}
+
+	if md, ok := metadata.FromIncomingContext(s.Context); ok {
+		return context.FromContext(metadata.NewOutgoingContext(context.Background(), md))
+	}
+	return context.Background()
+}
+
+func newSendVerifyStorageChallengeMsg(ctx context.Context, verifierMasternodesAddr []string, challengeMsg *ChallengeMessage) *sendVerifyStorageChallengeMsg {
+	return &sendVerifyStorageChallengeMsg{
+		CommonProtoMsg:          &messaging.CommonProtoMsg{},
+		Context:                 ctx,
+		VerifierMasternodesAddr: verifierMasternodesAddr,
+		ChallengeMessage:        challengeMsg,
+	}
+}
+
+type sendProcessStorageChallengeMsg struct {
+	*messaging.CommonProtoMsg
+	Context                  context.Context
+	ProcessingMasternodeAddr string
+	*ChallengeMessage
+}
+
+func (s *sendProcessStorageChallengeMsg) getAppContext() context.Context {
+	if s.Context == nil {
+		return context.Background()
+	}
+
+	if md, ok := metadata.FromIncomingContext(s.Context); ok {
+		return context.FromContext(metadata.NewOutgoingContext(context.Background(), md))
+	}
+	return context.Background()
+}
+
+func newSendProcessStorageChallengeMsg(ctx context.Context, processingMasternodeAddr string, challengeMsg *ChallengeMessage) *sendProcessStorageChallengeMsg {
+	return &sendProcessStorageChallengeMsg{
+		CommonProtoMsg:           &messaging.CommonProtoMsg{},
+		Context:                  ctx,
+		ProcessingMasternodeAddr: processingMasternodeAddr,
+		ChallengeMessage:         challengeMsg,
+	}
+}
+
+type domainActor struct {
+	conn node.Client
+}
+
+func (d *domainActor) Receive(context actor.Context) {
+	switch msg := context.Message().(type) {
+	case *sendVerifyStorageChallengeMsg:
+		d.OnSendVerifyStorageChallengeMessage(msg)
+	case *sendProcessStorageChallengeMsg:
+		d.OnSendProcessStorageChallengeMessage(msg)
+	default:
+		log.WithField("actor", "domain actor").Debug("Unhandled action", msg)
+	}
+}
+
+func newDomainActor(conn node.Client) actor.Actor {
+	return &domainActor{
+		conn: conn,
+	}
+}
+
+// OnSendVerifyStorageChallengeMessage handle event sending verity storage challenge message
+func (d *domainActor) OnSendVerifyStorageChallengeMessage(msg *sendVerifyStorageChallengeMsg) {
+	appCtx := msg.getAppContext()
+	for _, verifyingMasternodeAddr := range msg.VerifierMasternodesAddr {
+		log.Debug(verifyingMasternodeAddr)
```

Comment:
could you add more context info to Debug message?
like log.WithField("verifyingMasternodeAddr", verifyingMasternodeAddr).Debug("OnSendVerifyStorageChallengeMessage")?